### PR TITLE
Add `from_lists` to `TextClassificationData`

### LINKED ISSUE #769: Add support for a list of strings and list of targets to text classification
## 🚀 Feature

We should support something like we have for images with the `from_files`, but instead with a list of strings and a list of targets.

### PR DESCRIPTION
**Pull request by kingyiusuen**
## What does this PR do?

Fixes #769 (issue)

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/text/classification/data.py [hunk 1]
```diff
@@ -278,6 +278,49 @@ def __setstate__(self, state):
...
+        self,
+        data: Tuple[List[str], List[Any]],
+        dataset: Optional[Any] = None,
+        columns: Union[List[str], Tuple[str]] = ("input_ids", "attention_mask", "labels"),
+    ) -> Union[Sequence[Mapping[str, Any]]]:
+        input, target = data
+        hf_dataset = Dataset.from_dict({"input": input, "labels": target})
+
+        if not self.predicting:
+            dataset.multi_label = False
```
Comment (tchaton) [inline flash/text/classification/data.py:292]: Would you mind adding support for multi label ?

File: flash/text/classification/data.py [hunk 2]
```diff
@@ -450,3 +494,78 @@ def from_data_frame(
...
+        list.
+
+        Args:
+            train_data: A list to use as the train inputs.
+            train_targets: A sequence of targets (one per train input) to use as the train targets.
+            val_data: A list to use as the validation inputs.
+            val_targets: A sequence of targets (one per validation input) to use as the validation targets.
+            test_data: A list to use as the test inputs.
+            test_targets: A sequence of targets (one per test input) to use as the test targets.
+            predict_data: A list to use when predicting.
```
Comment (tchaton) [inline flash/text/classification/data.py:530]:
```suggestion
            train_data: A list sentences to use as the train inputs.
            train_targets: A sequence of targets (one per train input) to use as the train targets.
            val_data: A list sentences to use as the validation inputs.
            val_targets: A sequence of targets (one per validation input) to use as the validation targets.
            test_data: A list sentences to use as the test inputs.
            test_targets: A sequence of targets (one per test input) to use as the test targets.
            predict_data: A list sentences to use when predicting.
```

File: flash/text/classification/data.py [hunk 3]
```diff
@@ -330,6 +373,7 @@ def __init__(
                 DefaultDataSources.CSV: TextCSVDataSource(self.backbone, max_length=max_length),
                 DefaultDataSources.JSON: TextJSONDataSource(self.backbone, max_length=max_length),
                 "data_frame": TextDataFrameDataSource(self.backbone, max_length=max_length),
+                "list": TextListDataSource(self.backbone, max_length=max_length),
```
Comment (tchaton) [inline flash/text/classification/data.py:376]: Let's create a List within DefaultDataSources enum :) Same for DataFrame and Sentences

File: flash/text/classification/data.py (hunk 1 above)
Comment (kingyiusuen) [reply flash/text/classification/data.py:292]: No problem :) What should `targets` look like? A list of `num_samples` lists, each has `num_classes` items? Or a list of `num_classes` lists, each has `num_samples` items?

**Review by tchaton — APPROVED** [review]
LGTM ! Great work !

**Review by SeanNaren — APPROVED** [review]

**Comment by SeanNaren** [comment]
Thank you @kingyiusuen!!
